src/kanata/debounce/test/sym_defer_pk.rs

```diff
@@ -0,0 +1,114 @@
+        let has_pending_after_tick = algorithm.tick(&tx, Instant::now());
+        assert!(!has_pending_after_tick, "Expected no pending events after tick");
+
+        // Verify the event was sent
+        let received_event = rx.try_recv().expect("Expected an event");
+        assert_eq!(received_event.code, key_event.code);
+        assert_eq!(received_event.value, key_event.value);
+    }
+
+    #[test]
+    fn debounce_key_press_and_release() {
+        let (tx, rx) = mpsc::sync_channel(10);
+        let mut algorithm = create_debounce_algorithm(DebounceAlgorithm::SymDeferPk, 50);
+
+        let key_press = KeyEvent::new(OsCode::KEY_A, KeyValue::Press);
+        let key_release = KeyEvent::new(OsCode::KEY_A, KeyValue::Release);
+
+        // Process key press
+        algorithm.process_event(key_press, &tx);
+        assert!(rx.try_recv().is_err(), "Expected no key press event immediately");
+
+        // Process key release
+        algorithm.process_event(key_release, &tx);
+        assert!(rx.try_recv().is_err(), "Expected no key release event immediately, this one should be debounced");
+
+        // Simulate a tick after debounce duration
+        std::thread::sleep(std::time::Duration::from_millis(51));
+        algorithm.tick(&tx, Instant::now());
+
+        // Verify key press event
+        let press_event = rx.try_recv().expect("Expected a key press event");
+        assert_eq!(press_event.code, key_press.code);
+        assert_eq!(press_event.value, key_press.value);
+
+        // Release should have been debounced
+        assert!(rx.try_recv().is_err(), "Expected no key release after debounce");
+    }
+
+    #[test]
+    fn test_repeat_event() {
+        let (tx, rx) = mpsc::sync_channel(10);
+        let mut algorithm = create_debounce_algorithm(DebounceAlgorithm::SymDeferPk, 50);
+
+        let repeat_event = KeyEvent::new(OsCode::KEY_A, KeyValue::Repeat);
+
+        // Process repeat event
+        let has_pending = algorithm.process_event(repeat_event, &tx);
+        assert!(!has_pending, "Repeat events should not create pending deadlines");
+
+        // Verify the repeat event was sent immediately
+        let received_event = rx.try_recv().expect("Expected a repeat event");
+        assert_eq!(received_event.code, repeat_event.code);
+        assert_eq!(received_event.value, repeat_event.value);
+    }
+
+    #[test]
+    fn test_multiple_keys() {
+        let (tx, rx) = mpsc::sync_channel(10);
+        let mut algorithm = create_debounce_algorithm(DebounceAlgorithm::SymDeferPk, 50);
+
+        let key_a_press = KeyEvent::new(OsCode::KEY_A, KeyValue::Press);
+        let key_b_press = KeyEvent::new(OsCode::KEY_B, KeyValue::Press);
+
+        // Process key A press
+        algorithm.process_event(key_a_press, &tx);
+        assert!(rx.try_recv().is_err(), "Expected no key A press event immediately");
+
+        std::thread::sleep(std::time::Duration::from_millis(1)); // Simulate a short delay
```

Comment:
In this test I noticed that the sym_defer_pk algorithm, as implemented right now, can change the order of the pressed keys if they are pressed at exactly the same time.

This can be verified by removing the `Simulate a short delay` line.

I am not sure if this is an issue. But feel free to comment @jtroo since you have much more experience in these matters.

Reply:
Would this be because the iteration order of `HashMap` is not guaranteed? 
```
        for (&oscode, &(ref event, deadline)) in &self.pending_events {
            if now >= deadline {
                log::debug!("Emitting deferred event for {:?}", oscode);
                try_send_panic(process_tx, event.clone());
                to_remove.push(oscode);
            }
        }
```